qiskit_machine_learning/neural_networks/sampler_qnn.py

```diff
@@ -0,0 +1,296 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2020, 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""A Neural Network implementation based on the Sampler primitive."""
+
+from __future__ import annotations
+import logging
+from numbers import Integral
+from typing import Optional, Union, List, Tuple, Callable, cast, Iterable
+
+import numpy as np
+
+from qiskit import QuantumCircuit
+from qiskit.circuit import Parameter
+from qiskit.primitives import BaseSampler
+from qiskit.algorithms.gradients import BaseSamplerGradient
+from qiskit_machine_learning.exceptions import QiskitMachineLearningError, QiskitError
+from .neural_network import NeuralNetwork
+
+logger = logging.getLogger(__name__)
+
+
+class SamplerQNN(NeuralNetwork):
+    """A Neural Network implementation based on the Sampler primitive."""
+
+    def __init__(
+        self,
+        sampler: BaseSampler,
+        circuit: QuantumCircuit,
+        input_params: List[Parameter] | None = None,
+        weight_params: List[Parameter] | None = None,
+        interpret: Callable[[int], int | Tuple[int, ...]] | None = None,
+        output_shape: int | Tuple[int, ...] | None = None,
+        gradient: BaseSamplerGradient | None = None,
+        input_gradients: bool = False
+    )-> None:
+        """
+
+        Args:
+            circuit: The parametrized quantum circuit that generates the samples of this network.
+            input_params: The parameters of the circuit corresponding to the input.
+            weight_params: The parameters of the circuit corresponding to the trainable weights.
```

Comment:
I think the descriptions here could include what happens for None (default) case.